src/RunGridsearchCV.py

```diff
@@ -0,0 +1,128 @@
+# pipeline components
+from src.preprocess.FeatureExtractor import FeatureExtractor
+from src.preprocess.MissingFeaturesImputer import MissingFeaturesImputer
+from src.preprocess.SelectKBest import SelectKBest
+from src.preprocess.oversampler import Oversampler
+
+import pandas as pd
+from imblearn.pipeline import Pipeline
+from sklearn.model_selection import GridSearchCV
+
+# models
+from sklearn.svm import SVC
+from sklearn.ensemble import RandomForestClassifier, AdaBoostClassifier, GradientBoostingClassifier
+from xgboost import XGBClassifier
+
+import time
+import os
+
+
+# TODO: add a graph to compare the performance of the models over different metrics
+
+
+def run_all_models(func):
+    def run_loop(**kargs):
+        model_names = kargs.pop('model_names')
+        if os.path.exists(kargs['out_path']):
+            os.remove(kargs['out_path'])
+
+        for model_name in model_names:
+            print(f'Running {model_name}:')
+            func(model_name=model_name, **kargs)
+
+    return run_loop
+
+
+def runtime_counter(func):
+    def timer(**kargs):
+        start_time = time.time()
+        func(**kargs)
+        end_time = time.time()
+
+        time_diff = end_time - start_time
+        minutes = int(time_diff // 60)
+        seconds = int(time_diff % 60)
+        print(f'Total run time: {minutes} minutes and {seconds} seconds\n')
+
+    return timer
+
+
+def set_logger(func):
+    def logger(**kargs):
+        out_path = kargs.pop('out_path')
+        model_name = kargs['model_name']
```

Comment:
This functions shouldn't be in this file... those are pipe utilities and should be in a file on their own...